tests/frame/with_row_index_test.py

```diff
@@ -1,20 +1,66 @@
 from __future__ import annotations
 
+from contextlib import nullcontext as does_not_raise
+from typing import TYPE_CHECKING
+
 import pytest
 
 import narwhals as nw
-from tests.utils import Constructor, assert_equal_data
+import narwhals.stable.v1 as nw_v1
+from tests.utils import Constructor, ConstructorEager, assert_equal_data
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from types import ModuleType
+
+data = {"abc": ["foo", "bars"], "xyz": [100, 200], "const": [42, 42]}
 
-data = {"a": ["foo", "bars"], "ab": ["foo", "bars"]}
+
+def test_with_row_index_eager(constructor_eager: ConstructorEager) -> None:
+    result = nw.from_native(constructor_eager(data), eager_only=True).with_row_index()
+    expected = {"index": [0, 1], **data}
+    assert_equal_data(result, expected)
 
 
-def test_with_row_index(constructor: Constructor, request: pytest.FixtureRequest) -> None:
-    if (
-        ("pyspark" in str(constructor))
-        or "duckdb" in str(constructor)
-        or "ibis" in str(constructor)
-    ):
-        request.applymarker(pytest.mark.xfail)
-    result = nw.from_native(constructor(data)).with_row_index()
-    expected = {"index": [0, 1], "a": ["foo", "bars"], "ab": ["foo", "bars"]}
+@pytest.mark.parametrize(
+    ("order_by", "expected_index"),
+    [
+        ("abc", [1, 0]),
+        ("xyz", [0, 1]),
+        (["const", "abc"], [1, 0]),
+        (["const", "xyz"], [0, 1]),
+    ],
+)
+def test_with_row_index_lazy(
+    constructor: Constructor, order_by: str | Sequence[str], expected_index: list[int]
+) -> None:
+    result = (
+        nw.from_native(constructor(data))
+        .with_row_index(name="foo bar", order_by=order_by)
+        .sort("xyz")
+    )
+    expected = {"foo bar": expected_index, **data}
     assert_equal_data(result, expected)
+
+
+@pytest.mark.parametrize("namespace", [nw, nw_v1])
+def test_with_row_index_lazy_exception(
+    constructor: Constructor, namespace: ModuleType
```

Comment:
really minor, but i think it'd be better to split this into a test in `tests/v1_test.py` too, so that `tests/v1_test.py` shows and tests all the places where behaviour differs